Plugins/SwiftGRPCPlugin/plugin.swift

```diff
@@ -0,0 +1,189 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Foundation
+import PackagePlugin
+
+@main
+struct SwiftGRPCPlugin: BuildToolPlugin {
+  /// Errors thrown by the `SwiftGRPCPlugin`
+  enum PluginError: Error {
+    /// Indicates that the target where the plugin was applied to was not `SourceModuleTarget`.
+    case invalidTarget
+    /// Indicates that the file extension of an input file was not `.proto`.
+    case invalidInputFileExtension
+  }
+
+  /// The configuration of the plugin.
+  struct Configuration: Codable {
+    /// Encapsulates a single invocation of protoc.
+    struct Invocation: Codable {
+      /// The visibility of the generated files.
+      enum Visibility: String, Codable {
+        /// The generated files should have `internal` access level.
+        case `internal`
+        /// The generated files should have `public` access level.
+        case `public`
+      }
+
+      /// An array of paths to `.proto` files for this invocation.
+      var protoFiles: [String]
+      /// The visibility of the generated files.
+      var visibility: Visibility?
+      /// Whether server code is generated.
+      var server: Bool?
+      /// Whether client code is generated.
+      var client: Bool?
+      /// Whether test client code is generated.
+      var testClient: Bool?
+      /// Determines whether the casing of generated function names is kept.
+      var keepMethodCasing: Bool?
+    }
+
+    /// The path to the `protoc` binary.
+    ///
+    /// If this is not set, SPM will try to find the tool itself.
+    var protocPath: String?
+
+    /// A list of invocations of `protoc` with the `SwiftGRPCPlugin`.
+    var invocations: [Invocation]
+  }
+
+  static let configurationFileName = "swift-grpc-config.json"
+
+  func createBuildCommands(context: PluginContext, target: Target) throws -> [Command] {
+    // Let's check that this is a source target
+    guard let target = target as? SourceModuleTarget else {
+      throw PluginError.invalidTarget
+    }
+
+    // We need to find the configuration file at the root of the target
+    let configurationFilePath = target.directory.appending(subpath: Self.configurationFileName)
+    let data = try Data(contentsOf: URL(fileURLWithPath: "\(configurationFilePath)"))
+    let configuration = try JSONDecoder().decode(Configuration.self, from: data)
+
+    try self.validateConfiguration(configuration)
+
+    // We need to find the path of protoc and protoc-gen-swift
+    let protocPath: Path
+    if let configuredProtocPath = configuration.protocPath {
```

Comment:
This has changed on the latest swift-protobuf commit. If the env variable `PROTOC_PATH` is set we also try to respect that. Can you please take a look at how the latest implementation looks inside the protobuf repo and adopt it here.